File: enchantgui/gui/src/main/java/cc/mewcraft/enchantment/gui/command/PluginCommands.java
package cc.mewcraft.enchantment.gui.command;

import cc.mewcraft.enchantment.gui.EnchantGuiPlugin;
import cc.mewcraft.enchantment.gui.gui.EnchantMenu;
import cc.mewcraft.mewcore.command.SimpleCommands;
import cloud.commandframework.bukkit.parsers.PlayerArgument;
import com.google.inject.Inject;
import com.google.inject.Singleton;
import org.bukkit.entity.Player;

@Singleton
public class PluginCommands extends SimpleCommands<EnchantGuiPlugin> {
    @Inject
    public PluginCommands(final EnchantGuiPlugin plugin) throws Exception {
        super(plugin);
    }

    @Override public void prepareAndRegister() {
        // Prepare commands
        registry.prepareCommand(registry
            .commandBuilder("enchantgui")
            .literal("open")
            .permission("enchantgui.command.open")
            .argument(PlayerArgument.optional("target"))
            .handler(ctx -> {
                Player viewer;
                if (ctx.contains("target")) {
                    viewer = ctx.get("target");
                } else if (ctx.getSender() instanceof Player player) {
                    viewer = player;
                } else return;
                plugin.getInjector().getInstance(EnchantMenu.class).open(viewer);
            }).build());
        registry.prepareCommand(registry
            .commandBuilder("enchantgui")
            .literal("reload")
            .permission("enchantgui.command.reload")
            .handler(ctx -> {
                plugin.onDisable();
                plugin.onEnable();
            }).build());

        // Register commands
        registry.registerCommands();
    }
}

File: enchantgui/gui/src/main/kotlin/cc/mewcraft/enchantment/gui/command/PluginCommands.kt
package cc.mewcraft.enchantment.gui.command

import cc.mewcraft.enchantment.gui.EnchantGuiPlugin
import cc.mewcraft.enchantment.gui.gui.EnchantMenu
import cc.mewcraft.mewcore.command.SimpleCommands
import cloud.commandframework.bukkit.parsers.PlayerArgument
import com.google.inject.Inject
import com.google.inject.Singleton
import org.bukkit.entity.Player

@Singleton
class PluginCommands @Inject constructor(
    plugin: EnchantGuiPlugin,
) : SimpleCommands<EnchantGuiPlugin>(plugin) {
    override fun prepareAndRegister() {
        // Prepare commands
        registry.prepareCommand(
            registry
                .commandBuilder("enchantgui")
                .literal("open")
                .permission("enchantgui.command.open")
                .argument(PlayerArgument.optional("target"))
                .handler {

                    val viewer: Player = if (it.contains("target")) {
                        it.get("target")
                    } else if (it.sender is Player) {
                        it.sender as Player
                    } else return@handler

                    plugin.injector.getInstance(EnchantMenu::class.java).showMenu(viewer)

                }.build()
        )
        registry.prepareCommand(
            registry
                .commandBuilder("enchantgui")
                .literal("reload")
                .permission("enchantgui.command.reload")
                .handler {

                    plugin.onDisable()
                    plugin.onEnable()

                }.build()
        )

        // Register commands
        registry.registerCommands()
    }
}
